cmd/config.go

```diff
@@ -494,3 +494,47 @@ type OpenTelemetryConfig struct {
 	// decisions.
 	DisableParentSampler bool
 }
+
+// DNSProvider contains the configuration for a DNS provider in the bdns package
+// which supports dynamic reloading of its backends.
+type DNSProvider struct {
+	// DNSAuthority is a single <hostname|IPv4|[IPv6]>:<port> of the DNS server
+	// to be used for resolution of DNS backends. If the address contains a
+	// hostname it will be resolved via the system DNS. If the address contains
+	// a port, the client will use it directly, otherwise port 53 is assumed.
+	DNSAuthority string `validate:"omitempty,ip|hostname|hostname_port"`
+
+	// SRVLookup contains the service and domain name used to construct a SRV
+	// DNS query to lookup DNS backends. For example: if the resource record is
+	// 'unbound.service.consul', then the 'Service' is 'unbound' and the
+	// 'Domain' is 'service.consul'. The expected dNSName to be authenticated in
+	// the server certificate would be 'unbound.service.consul'.
+	//
+	// Note: The 'proto' field of the SRV record MUST contain 'udp' and the
+	// 'port' field MUST be a valid port. In a Consul configuration file you
+	// would specify 'unbound.service.consul' as:
+	//
+	// services {
+	//   id      = "unbound-1" // Must be unique
+	//   name    = "unbound"
+	//   address = "10.77.77.77"
+	//   port    = 53
+	//   tags    = ["udp"]
+	// }
+	//
+	// services {
+	//   id      = "unbound-2" // Must be unique
+	//   name    = "unbound"
+	//   address = "10.88.88.88"
+	//   port    = 53
+	//   tags    = ["udp"]
+	// }
+	//
+	// If you've added the above to your Consul configuration file (and reloaded
+	// Consul) then you should be able to resolve the following dig query:
+	//
+	// $ dig @10.55.55.10 -t SRV _unbound._udp.service.consul +short
+	// 1 1 53 0a585858.addr.dc1.consul.
+	// 1 1 53 0a4d4d4d.addr.dc1.consul.
+	SRVLookup ServiceDomain `validate:"required"`
```

Comment:
Two questions:
1) GRPCClientConfig uses `*ServiceDomain`. Any reason you're not using a pointer this time?
2) We ended up adding `SRVLookups []*ServiceDomain` to GRPCClientConfig for more flexibility. Should we just start that way here immediately?

Reply:
1.GRPCClientConfig uses `*ServiceDomain` because it being nilable makes working with it in `MakeTargetAndHostOverride()` waaaaay easier than checking if both inner strings == `""`.
2. That's a good idea, I'll kick that over to @jcjones and see what he has to say.

Reply:
Okay, @jcjones says, _"YAGNI."_